interface/src/transaction.rs

```diff
@@ -0,0 +1,34 @@
+pub use parity_scale_codec::{Decode, Encode};
+
+use crate::Address;
+
+#[derive(Debug, Clone, Encode, Decode)]
+pub struct Transaction {
```

Comment:
I wouldn't call this `Transaction`, I'd call it `TransactionPayload` or simply `Payload`. It's the inner part of the tx that's opaque to the host. It consists of just method and args.

- nonce will be handled by the host for now; we'll move it into the VM later
- principal needs to be readable by the host, so it sits outside this payload. It's the `sender` in the execution message.
- template isn't needed here; the template `code` is also part of the execution message.